java-client/src/main/java/com/cumulocity/sdk/client/ResponseParser.java

```diff
@@ -77,14 +83,29 @@ public void checkStatus(Response response, int... expectedStatusCodes) throws SD
 
     protected String getErrorMessage(Response response, int status) {
         String errorMessage = "Http status code: " + status;
+        String responseHeader = null;
 
         if (response.hasEntity()) {
             String errorRepresentation = getErrorRepresentation(response);
             if (errorRepresentation == null) {
                 errorRepresentation = NO_ERROR_REPRESENTATION;
             }
+            Map<String, List<String>> responseHeadermap = response.getStringHeaders();
+            if (responseHeadermap != null) {
+                for (Map.Entry<String, List<String>> entry : responseHeadermap.entrySet()) {
+                    responseHeader += entry.getKey() +
```

Comment:
My tests did not show any of those, but this may be due to poor test strategy.
Your approach is correct, to use those headers, that we are currently interrested is the better.